tests/std/tests/P0896R4_ranges_algorithm_machinery/test.cpp

```diff
@@ -533,3 +533,76 @@ namespace result_test {
     STATIC_ASSERT(test_convertible_3<in_out_out_result>());
 } // namespace result_test
+
+namespace permutable_test {
+    using std::forward_iterator, std::indirectly_movable_storable, std::indirectly_swappable, std::permutable;
+
+    template <int I>
+    struct archetype {
+        using value_type      = int;
+        using difference_type = int;
+
+        struct proxy {
+            proxy()             = default;
+            proxy(proxy const&) = delete;
+            proxy& operator=(proxy const&) = delete;
+
+            operator int() const;
+            // 1: not reflexively indirectly_movable_storable
+            void operator=(int) const requires(I != 1);
+        };
+
+        proxy operator*() const;
+
+        archetype& operator++();
+        archetype operator++(int);
+
+        // 0: not forward_iterator (input only)
+        bool operator==(archetype const&) const requires(I != 0) = default;
+
+        friend int iter_move(archetype const&) {
+            return 42;
+        }
+        friend void iter_swap(archetype const&, archetype const&) {}
+
+        // Ideally we'd have a "not reflexively indirectly_swappable" case as well, but there's no way for a class to
+        // satisfy indirectly_movable_storable<T, T> without satisfying indirectly_swappable<T, T> thanks to N4861
+        // [iterator.cust.swap]/4.3. permutable requires indirectly_swappable<T, T> only to ensure that a user type
+        // doesn't define an iter_swap overload that doesn't meet the semantic requirements.
+    };
+    STATIC_ASSERT(std::input_iterator<archetype<0>>);
+    STATIC_ASSERT(!std::forward_iterator<archetype<0>>);
+    STATIC_ASSERT(indirectly_movable_storable<archetype<0>, archetype<0>>);
+    STATIC_ASSERT(indirectly_swappable<archetype<0>, archetype<0>>);
+
+    STATIC_ASSERT(std::forward_iterator<archetype<1>>);
+    STATIC_ASSERT(!indirectly_movable_storable<archetype<1>, archetype<1>>);
+    STATIC_ASSERT(indirectly_swappable<archetype<1>, archetype<1>>);
+
+    STATIC_ASSERT(std::forward_iterator<archetype<2>>);
+    STATIC_ASSERT(indirectly_movable_storable<archetype<2>, archetype<2>>);
+    STATIC_ASSERT(indirectly_swappable<archetype<2>, archetype<2>>);
+
+    STATIC_ASSERT(!permutable<archetype<0>>);
+    STATIC_ASSERT(!permutable<archetype<1>>);
+    STATIC_ASSERT(permutable<archetype<2>>);
+} // namespace permutable_test
+
+namespace sortable_test {
+    using ranges::less;
+    using std::identity, std::indirect_strict_weak_order, std::permutable, std::projected, std::sortable;
+
+    // not permutable
+    STATIC_ASSERT(!permutable<int const*>);
+    STATIC_ASSERT(indirect_strict_weak_order<less, projected<int const*, identity>>);
+    STATIC_ASSERT(!sortable<int const*, less, identity>);
+
+    // not indirect_strict_weak_order
+    STATIC_ASSERT(permutable<int*>);
+    STATIC_ASSERT(!indirect_strict_weak_order<void, projected<int const*, identity>>);
+    STATIC_ASSERT(!sortable<int*, int, identity>);

```

Comment:
Did you even look at these? They don't agree. +100 to the above comment about symbolic names.
```suggestion
    STATIC_ASSERT(!sortable<int*, void, identity>);
```